lib/spack/spack/build_systems/_checks.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import os
+
+import six
+
+import llnl.util.lang
+
+import spack.builder
+import spack.installer
+import spack.relocate
+import spack.store
+
+
+def sanity_check_prefix(builder):
```

Comment:
Because:
1. With the changes in the class structure a callback requires a builder, like in the signature above
2. `PackageBase` didn't seem the right place for the function given that signature

Reply:
I thought callbacks *could* take a package, like in `var/spack/repos/builtin.mock/packages/cmake-client/package.py`: if the callback is defined in the package, it takes the package as the first (and only) arg, and likewise if defined in the builder it takes the builder as the first (and only) arg.

Reply:
> I got that impression from the docstring of combine_callbacks:

The package adapter is only used when we create the adapter. So I would present it like that:
1. The callbacks always take a builder
2. When we create an adapter for single-class recipes, we take care of plugging in callbacks defined on the package that take a  package

Reply:
Oh whoops yes this adapter is only used for classes that do not define their own builder. Is it assumed then that no existing user-defined package uses any of these callbacks (or in the case of `execute_build_time_tests ` which was renamed from `_run_default_build_time_test_callbacks`, no packages refer to `_run_default_build_time_test_callbacks`)? Any package that did would not be backwards compatible. I don't see any packages in the builtin repository that do this, and it also seems unlikely that anyone would refer to these callbacks (they either apply to all packages or are indicated as being private).

Reply:
> Is it assumed then that no existing user-defined package uses any of these callbacks

Yes, as the name started with a leading underscore I assumed it was safe to extract them from the class. I also didn't find any use in the builtin repository.